packages/components/src/test-utils/addParentAndSetRequiredProps.ts

```diff
@@ -22,27 +21,35 @@ export const addParentAndSetRequiredProps = (tagName: TagName, component: any):
     component[childTagName] = child;
   }
 
-  if (requiredNamedSlots) {
-    requiredNamedSlots.forEach(({ slotName, tagName: requiredTagName }) => {
-      const child = document.createElement(requiredTagName);
+  const requiredNamedSlots = Object.entries(slotsMeta ?? {}).filter(([, value]) => value.isRequired);
+
+  if (requiredNamedSlots.length > 0) {
+    requiredNamedSlots.forEach(([slotName, value]) => {
+      const child = document.createElement(value.allowedTagNames[0]);
       child.slot = slotName;
 
-      if (requiredTagName.includes('link')) {
+      if (value.allowedTagNames[0].includes('link')) {
         (child as any).href = '#';
       }
 
       component.host.appendChild(child);
     });
   }
 
+  const requiredProps = propsMeta
+    ? Object.entries(propsMeta)
+        .filter(([, value]) => value.isRequired)
+        .map(([key]) => key)
+    : undefined;
```

Comment:
`propsMeta` is fallbacked by `{}` in line 5, when no value is passed, the filter will be run on an empty Object. This could be avoided. Also ternaries add complexity, better write it out for more readability.